### PR TITLE
fix(docs-ui): formula-editor can't scroll & edit link on cell editor

### PR DESCRIPTION
**Pull request by weird94**
close https://github.com/dream-num/univer-pro/issues/2513
close https://github.com/dream-num/univer-pro/issues/2591
close https://github.com/dream-num/univer-pro/issues/2427
close https://github.com/dream-num/univer-pro/issues/1037
close https://github.com/dream-num/univer-pro/issues/2589
close https://github.com/dream-num/univer-pro/issues/1065

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

**Comment by zhaolixin7** [comment]
#2513。@weird94 添加定位到单元格直接选择选区添加链接失败
[2024_9_23 17_09_00 commands.json](https://github.com/user-attachments/files/17095011/2024_9_23.17_09_00.commands.json)
[2024_9_23 17_09_00 video.webm](https://github.com/user-attachments/assets/0312526e-fcee-40cd-91d1-bdc4a589fe86)

**Comment by zhaolixin7** [comment]
#1065  路径：在合并单元格（sheet-0004 h1）编辑后再编辑其他单元格
 表现：编辑框异常
[2024_9_24 10_59_21 video.webm](https://github.com/user-attachments/assets/a7f6821e-044e-4cef-86a0-1ca4c2f76db7)
image

**Comment by weird94** [comment]
> #1065 路径：在合并单元格（sheet-0004 h1）编辑后再编辑其他单元格 表现：编辑框异常 [2024_9_24 10_59_21 video.webm](https://github.com/user-attachments/assets/a7f6821e-044e-4cef-86a0-1ca4c2f76db7) image

这个我没法复现，其他的我改掉了

File: packages/docs-ui/src/services/selection/doc-selection-render.service.ts [hunk 1]
```diff
@@ -127,7 +127,7 @@ export class DocSelectionRenderService extends RxDisposable implements IRenderMo
 
                 const unitId = documentModel.getUnitId();
 
-                if (unitId !== this._context.unitId && !this._reserveRanges) {
+                if (unitId !== this._context.unitId) {
```
Comment (Jocs) [inline packages/docs-ui/src/services/selection/doc-selection-render.service.ts:130]: 这个为啥删除呢？

File: packages/docs-ui/src/services/selection/doc-selection-render.service.ts [hunk 2]
```diff
@@ -721,7 +721,9 @@ export class DocSelectionRenderService extends RxDisposable implements IRenderMo
     }
 
     private _removeAllRanges() {
-        this._removeAllTextRanges();
+        if (!this._reserveRanges) {
```
Comment (Jocs) [inline packages/docs-ui/src/services/selection/doc-selection-render.service.ts:724]: 不能直接添加到这儿吧？不然有耦合了

**Review by Jocs — APPROVED** [review]

**Pull request merged**